pyvista/utilities/reader.py

```diff
@@ -980,6 +998,282 @@ class Plot3DMetaReader(BaseReader):
+        zone, set this to ``False``. The default is ``True``.
+
+        Returns
+        -------
+        bool
+            If ``True``, distribute each block in each zone across ranks.
+
+        Examples
+        --------
+        Disable distributing blocks.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.distribute_blocks = False
+        >>> reader.distribute_blocks
+        False
+
+        """
+        return bool(self._reader.GetDistributeBlocks())
+
+    @distribute_blocks.setter
+    def distribute_blocks(self, value: str):
+        self._reader.SetDistributeBlocks(value)
+
+    def base_array_status(self, name: str) -> bool:
+        """Get status of base array with name.
+
+        Parameters
+        ----------
+        name : str
+            Base array name.
+
+        Returns
+        -------
+        bool
+            Whether reading the base array is enabled.
+
+        """
+        return bool(self.reader.GetBaseArrayStatus(name))
+
+    @property
+    def base_array_names(self):
+        """Return the list of all base array names.
+
+        Returns
+        -------
+        list[int]
+
+        """
+        return [self.reader.GetPointArrayName(i) for i in range(self.number_point_arrays)]
+
+    def enable_all_bases(self):
+        """Enable reading all bases.
+
+        By default only the 0th base is read.
+
+        Examples
+        --------
+        Read all bases.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.enable_all_bases()
+        """
+        self._reader.EnableAllBases()
+
+    def disable_all_bases(self):
+        """Disable reading all bases.
+
+        By default only the 0th base is read.
+
+        Examples
+        --------
+        Disable reading all bases.
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cgns_multi(load=False)
+        >>> reader = pyvista.get_reader(filename)
+        >>> reader.disable_all_bases()
+        """
+        self._reader.DisableAllBases()
+
+    def family_array_status(self, name) -> bool:
+        """Get status of family array with name.
+
+        Parameters
+        ----------
+        name : str
+            Family array name.
+
+        Returns
+        -------
+        bool
+            Whether reading the family array is enabled.
+
+        """
+        return bool(self.reader.GetFamilyArrayStatus(name))
+
+    @property
+    def family_array_names(self) -> List[str]:
+        """Return the list of all family array names.
+
+        Returns
+        -------
+        list[str]
+
+        """
+        return [self.reader.GetPointArrayName(i) for i in range(self.number_point_arrays)]
```

Comment:
Is this supposed to be `GetFamilyArrayName(i)` and correspondingly, the number of family arrays?

Reply:
Correct. Fixed in f315558.